sdk/src/test/scala/com/horizen/account/block/AccountBlockHeaderTest.scala

```diff
@@ -0,0 +1,173 @@
+
+    val serializedHeader = serializedHeaderTry.get
+    assertEquals("AccountBlockHeader version is different", header.version, serializedHeader.version)
+    assertEquals("AccountBlockHeader parentId is different", header.parentId, serializedHeader.parentId)
+    assertEquals("AccountBlockHeader timestamp is different", header.timestamp, serializedHeader.timestamp)
+    assertEquals("AccountBlockHeader forgingStakeInfo is different", header.forgingStakeInfo, serializedHeader.forgingStakeInfo)
+    assertEquals("AccountBlockHeader forgingStakeMerklePath is different", header.forgingStakeMerklePath, serializedHeader.forgingStakeMerklePath)
+    assertArrayEquals("AccountBlockHeader vrfProof is different", header.vrfProof.bytes, serializedHeader.vrfProof.bytes) // TODO: replace with vrfProof inself later
+    assertArrayEquals("AccountBlockHeader sidechainTransactionsMerkleRootHash is different", header.sidechainTransactionsMerkleRootHash, serializedHeader.sidechainTransactionsMerkleRootHash)
+    assertArrayEquals("AccountBlockHeader mainchainMerkleRootHash is different", header.mainchainMerkleRootHash, serializedHeader.mainchainMerkleRootHash)
+    assertArrayEquals("AccountBlockHeader stateRoot is different", header.stateRoot, serializedHeader.stateRoot)
+    assertArrayEquals("AccountBlockHeader receiptsRoot is different", header.receiptsRoot, serializedHeader.receiptsRoot)
+    assertEquals("AccountBlockHeader forgerAddress is different", header.forgerAddress, serializedHeader.forgerAddress)
+    assertEquals("AccountBlockHeader baseFee is different", header.baseFee, serializedHeader.baseFee)
+    assertEquals("AccountBlockHeader gasUsed is different", header.gasUsed, serializedHeader.gasUsed)
+    assertEquals("AccountBlockHeader gasLimit is different", header.gasLimit, serializedHeader.gasLimit)
+    assertEquals("AccountBlockHeader ommersNumber is different", header.ommersCumulativeScore, serializedHeader.ommersCumulativeScore)
+    assertEquals("AccountBlockHeader id is different", header.id, serializedHeader.id)
+
+    // Set to true to regenerate regression data
+    if(false) {
+      val out = new BufferedWriter(new FileWriter("src/test/resources/accountblockheader_hex"))
+      out.write(BytesUtils.toHexString(bytes))
+      out.close()
+    }
+
+    // Test 2: try to deserialize broken bytes.
+    assertTrue("AccountBlockHeaderSerializer expected to be not parsed due to broken data.", AccountBlockHeaderSerializer.parseBytesTry("broken bytes".getBytes).isFailure)
+  }
+
+  @Test
+  def serializationRegression(): Unit = {
+    var bytes: Array[Byte] = null
+    try {
+      val classLoader = getClass.getClassLoader
+      val file = new FileReader(classLoader.getResource("accountblockheader_hex").getFile)
+      bytes = BytesUtils.fromHexString(new BufferedReader(file).readLine())
+    }
+    catch {
+      case e: Exception =>
+        jFail(e.toString)
```

Comment:
Consider to avoid catching the exception, because if the exception is caught by the test framework also the stack trace is printed, while in this case you print just the exception message

Reply:
Done